clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -10619,36 +10608,66 @@ static void getOtherSYCLPostLinkOpts(const ToolChain &TC, const JobAction &JA,
 }
 
 // Add any sycl-post-link options that rely on a specific Triple.
-static void
-getTripleBasedSYCLPostLinkOpts(const ToolChain &TC, const JobAction &JA,
-                               const llvm::opt::ArgList &TCArgs,
-                               llvm::Triple Triple, ArgStringList &PostLinkArgs,
-                               bool SpecConsts, types::ID OutputType) {
+static void getTripleBasedSYCLPostLinkOpts(
+    const ToolChain &TC, const JobAction &JA, const llvm::opt::ArgList &TCArgs,
+    llvm::Triple Triple, ArgStringList &PostLinkArgs, bool SpecConsts,
+    types::ID OutputType, bool NewOffloadDriver = false) {
```

Comment:
Instead of a parameter for `NewOffloadDriver` why not use the `getUseNewOffloadingDriver()`?

Reply:
`TC.getDriver()` might work